plugins/permission-react/src/components/RequirePermission.tsx

```diff
@@ -54,14 +56,14 @@ export const PermissionedRoute = (
   const app = useApp();
   const { NotFoundErrorPage } = app.getComponents();
 
-  let shownElement: ReactElement | null | undefined =
-    errorComponent === undefined ? <NotFoundErrorPage /> : errorComponent;
-
   if (permissionResult.loading) {
-    shownElement = null;
-  } else if (permissionResult.allowed) {
-    shownElement = props.element;
+    return null;
   }
-
-  return <Route {...otherProps} element={shownElement} />;
-};
+  const unauthorizedResult =
+    unauthorizedComponent === undefined ? (
+      <NotFoundErrorPage />
```

Comment:
`PermissionedRoute` clearly reminds me to something related to Routes, while the new name of this component is more generic and can be used everywhere. For this reason maybe we shouldn't use `NotFoundErrorPage` here, unless we change the name of the component to something more specific, like `RequirePermissionRoute`